src/Verifiers/Dns01.php

```diff
@@ -49,13 +49,13 @@ public function verifyChallenge(string $domain, string $expectedPayload): void
         $uri = '_acme-challenge.' . $domain;
 
         try {
-            $dnsRecords = $this->resolver->query($uri, Dns\Record::TXT);
-        } catch (Dns\NoRecordException $e) {
-            throw new AcmeException("Verification failed, no TXT record found for '{$uri}'.", 0, $e);
+            $dnsRecords = $this->resolver->query($uri, Dns\DnsRecord::TXT);
+        } catch (Dns\MissingDnsRecordException $e) {
+            throw new AcmeException("Verification failed, no TXT record found for '{$uri}'.", "0", $e);
         } catch (Dns\DnsException $e) {
             throw new AcmeException(
                 "Verification failed, couldn't query TXT record of '{$uri}': " . $e->getMessage(),
-                0,
+                "0",
```

Comment:
We should probably have a separate look at this change, the type seems correct, but I'm not sure whether we should provide `0` in that case.

Reply:
I changed it to first try to use the code of the thrown DnsException or else use a 404 in case of a `MissingDnsRecordException` or a 500 otherwise. Would that make sense?